### PR TITLE
System xml cleanup

### PR DESCRIPTION
**Pull request by Bartlomiejsz**
### Description (*)

This PR aims to (at least partially) cleanup system.xml files:
- `showInStore="0"`, `showInWebsite="0"`, `showInDefault="0"` entries were removed as this is default value
- some default values were added to config.xml (i.e. no value set for field with `Magento\Config\Model\Config\Source\Yesno` model was set to 0 as default (to not change current behavior)
- more fields have `canRestore="1"` set now, to make it possible to reset to default value
- more configuration sections for one specific functionality have `depends` configuration added to only show if functionality itself is enabled.

This also includes changes from #26354 - the other PR can be merged at first or just closed if necessary.

### Manual testing scenarios (*)

1. Go to `Stores -> Configuration`
2. More fields (too many changes to include every single place changed here) should have checkbox to restore default value
3. More functionality configurations should be visible only if functionality is enabled (i.e. Newsletter)

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml [hunk 1]
```diff
@@ -16,7 +16,8 @@
                        sortOrder="80"
                        showInDefault="1"
                        showInWebsite="1"
-                       showInStore="1">
+                       showInStore="1"
+                       canRestore="1">
```
Comment (lbajsarowicz) [inline app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml]: You can also make it oneliner.

File: app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml [hunk 2]
```diff
@@ -27,7 +28,8 @@
                        sortOrder="81"
                        showInDefault="1"
                        showInWebsite="1"
-                       showInStore="1">
+                       showInStore="1"
+                       canRestore="1">
```
Comment (lbajsarowicz) [inline app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml]: You can also make it oneliner.

File: app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml [hunk 3]
```diff
@@ -40,27 +42,28 @@
                        sortOrder="82"
                        showInDefault="1"
                        showInWebsite="1"
-                       showInStore="1">
+                       showInStore="1"
+                       canRestore="1">
```
Comment (lbajsarowicz) [inline app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml]: You can also make it oneliner.

File: app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml [hunk 4]
```diff
@@ -40,27 +42,28 @@
...
                     <label>Show Results Count for Each Recommendation</label>
                     <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
                     <depends>
                         <field id="search_recommendations_enabled">1</field>
                     </depends>
                 </field>
                 <!--<group id="suggestions">-->
-                    <field id="search_suggestion_enabled" translate="label comment" type="select" sortOrder="90" showInDefault="1" showInWebsite="1" showInStore="1">
+                    <field id="search_suggestion_enabled" translate="label comment" type="select" sortOrder="90" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
```
Comment (lbajsarowicz) [inline app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml:33]: Nice catch with `canRestore` as default value is available in `config.xml` file

https://github.com/magento/magento2/blob/b75b478d20827acd57494dc1ebc3bea40d0de3e7/app/code/Magento/AdvancedSearch/etc/config.xml#L12

**Review by lbajsarowicz — APPROVED** [review]
Well done!

@magento-engcom-team - I think we can add `complex` label, as it was done with a really big effort to make complex changes not only to `system.xml`, but also related `config.xml` files.

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6590](https://jira.corp.magento.com/browse/ENGCOM-6590) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

File: app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml (hunk 3 above)
Comment (Bartlomiejsz) [reply app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml]: @lbajsarowicz done

File: app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml (hunk 2 above)
Comment (Bartlomiejsz) [reply app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml]: @lbajsarowicz done

File: app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml (hunk 1 above)
Comment (Bartlomiejsz) [reply app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml]: @lbajsarowicz done

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Pull request merged**